client.go

```diff
@@ -102,6 +102,9 @@ type Client struct {
 	subscriptions map[uint32]*Subscription
 	subMux        sync.RWMutex
 
+	//cancelMonitor cancels the monitorChannel goroutine
+	cancelMonitor context.CancelFunc
```

Comment:
it is normal to store a context related item in a struct?

Reply:
I hesitated with this as well but the alternative would likely be a done channel.  Using a done channel, the select statement grows by one case statement and has the same body as the ctx.Done() case.  This felt redundant so using the context seemed more natural as we were already using it. 

That said, I'm not opposed to changing it if there are strong arguments against it.

Reply:
I think this is OK. If the context would need to cancel more than just the monitor I'd suggest to make it more generic but this is good enough.